RevenueCatUI/Templates/Components/Packages/PurchaseButton/PurchaseButtonComponentView.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PurchaseButtonComponentView.swift
+//
+//  Created by Josh Holtz on 9/27/24.
+
+import Foundation
+import RevenueCat
+import SwiftUI
+
+#if PAYWALL_COMPONENTS
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct PurchaseButtonComponentView: View {
+
+    private let viewModel: PurchaseButtonComponentViewModel
+
+    internal init(viewModel: PurchaseButtonComponentViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        Button {
+            // WIP: Need to perform purchase logic
+        } label: {
+            // WIP: Need to add logic for intro offer
+            Text(viewModel.cta)
+                .font(viewModel.textStyle)
```

Comment:
I'd consider making two changes here for code reusability:
- Extract all of these modifiers to a shared view style
- Make them be view modifiers on the button itself, not the text, so that it can be reused for other buttons as well if needed in the future.

I'd probably make this be a custom `ButtonStyle` named something like `CTAButtonStyle` so it can be applied to other buttons consistently, like what Paul Hudson does here: https://www.hackingwithswift.com/quick-start/swiftui/customizing-button-with-buttonstyle

Then, for future buttons, we could do something like:

```swift
Button {
    // Do something
} label: {
    Text("Look at me, I'm a CTA Button!")
}
.buttonStyle(CTAButtonStyle())
```

Then, to inject the style values from the view model, you could make `CTAButtonStyle` accept a struct that takes in all the values it needs:

```swift
struct CTAButtonStyleConfig {
    let font: Font
    let fontWeight: FontWight
    // ...

    init(
        font: Font = .body,
        fontWeight: FontWeight = .regular,
        // ...
    )
}

Button {
    // Do something
} label: {
    Text("Look at me, I'm a CTA Button!")
}
.buttonStyle(CTAButtonStyle(
    .init(
        font: viewModel.font,
        fontWeight: viewModel.fontWeight,
        // ...
    )
))
```

This might be over-abstracting a bit, but I know one of my biggest pain points in Pines has been getting consistent label/button styling across the app. It can be a real pain to copy/paste all of those view modifiers everywhere!

Reply:
Ahhh, I always forget about `.buttonStyle()` 🙈 I need to use that in my own apps!

I don't know if this _exactly_ fits here because this is the only place where this is happening but... I could maybe do something similar since since this button and text and a few others might be setting this same group of properties. 

I might hold off on doing this until I see more of pattern but I really like how much cleaner that looks 😍